internal/driver/driver.go

```diff
@@ -51,16 +51,40 @@ func (d *Driver) createDeviceClient(info *ConnectionInfo) (DeviceClient, error)
 	return c, nil
 }
 
+func (d *Driver) disconnectDeviceClient(info *ConnectionInfo) error {
+	d.clientMutex.Lock()
+	defer d.clientMutex.Unlock()
+	key := info.String()
+	if _, ok := d.clientMap[key]; ok {
+		client := d.clientMap[key]
+		return client.CloseConnection()
+	}
+	return nil
+}
+
 func (d *Driver) removeDeviceClient(info *ConnectionInfo) {
 	d.clientMutex.Lock()
 	defer d.clientMutex.Unlock()
 	key := info.String()
-	delete(d.clientMap, key)
+	if _, ok := d.clientMap[key]; ok {
+		client := d.clientMap[key]
+		err := client.CloseConnection()
+		if err != nil {
+			driver.Logger.Errorf("Device client close error, client key = %s, err = %v", key, err)
+		} else {
+			driver.Logger.Infof("device client closed,client key = %s", key)
+		}
+		delete(d.clientMap, key)
+	}
 }
 
 func (d *Driver) DisconnectDevice(deviceName string, protocols map[string]models.ProtocolProperties) error {
```

Comment:
please help remove this function because it's a legacy code and nothing is calling this function

Reply:
ohh! was wondering why it was showing as unimplemented